app/src/main/java/com/android/sample/ui/authentication/SignUp.kt

```diff
@@ -56,101 +55,83 @@ fun SignUpScreen(navigationActions: NavigationActions) {
+            // Password field
+            PasswordTextField(
+                password = passwordState.value,
+                onPasswordChange = { passwordState.value = it },
+                isPasswordVisible = isPasswordVisible.value,
+                onPasswordVisibilityChange = { isPasswordVisible.value = !isPasswordVisible.value },
+                passwordError = passwordErrorState.value,
+            )
+            Spacer(modifier = Modifier.height(16.dp))
+          }
+          item {
+            // Sign up button
+            Button(
+                onClick = {
+                  when {
+                    !isValidEmail(emailState.value) -> {
+                      emailErrorState.value =
+                          "Please enter a valid email address" // Set the error message if email is
+                      // invalid
+                    }
+                    passwordState.value.isEmpty() -> {
+                      passwordErrorState.value =
+                          "Password cannot be empty" // Set the error message if password is empty
+                    }
+                    passwordState.value.length < 6 -> {
+                      passwordErrorState.value =
+                          "Password must be at least 6 characters long" // Set the error message for
+                      // short passwords
+                    }
+                    else -> {
+                      createUserWithEmailAndPassword(
+                          emailState.value,
+                          passwordState.value,
+                          context,
+                          onSuccess = { navigationActions.navigateTo(Screen.CREATE_PROFILE) })
+                    }
                   }
+                },
+                modifier = Modifier.fillMaxWidth(0.8f).height(48.dp).testTag("SignUpButton")) {
+                  Text("Sign up with Email", fontSize = 16.sp)
                 }
-              },
-              modifier = Modifier.fillMaxWidth(0.8f).height(48.dp).testTag("SignUpButton")) {
-                Text("Sign up with Email", fontSize = 16.sp)
-              }
-          Spacer(modifier = Modifier.height(16.dp))
-
-          // If user already has an account, navigate to the sign in screen
-          TextButton(
-              onClick = { navigationActions.navigateTo(Screen.AUTH) },
-              modifier = Modifier.fillMaxWidth(0.8f).height(36.dp).testTag("GoToSignInButton")) {
-                Text("Already an account?", fontSize = 16.sp)
-              }
+            Spacer(modifier = Modifier.height(16.dp))
+          }
+          item {
+            // If user already has an account, navigate to the sign in screen
+            TextButton(
+                onClick = { navigationActions.navigateTo(Screen.AUTH) },
+                modifier = Modifier.fillMaxWidth(0.8f).height(36.dp).testTag("GoToSignInButton")) {
+                  Text("Already an account?", fontSize = 16.sp)
+                }
+          }
```

Comment:
Same here, great use of a lazy column instead of a column to ensure smooth scrolling when too much components are on the screen